pkg/api/mock/mock_api_implementation.go

```diff
@@ -0,0 +1,239 @@
+package mock
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"reflect"
+	"strings"
+
+	"github.com/onsi/gomega/ghttp"
+	uuid "github.com/satori/go.uuid"
+	"go.anx.io/go-anxcloud/pkg/api"
+	"go.anx.io/go-anxcloud/pkg/api/types"
+	"go.anx.io/go-anxcloud/pkg/client"
+)
+
+const (
+	// CloudDNS
+	resourcePrefixCloudDNSZone = "/api/clouddns/v1/zone.json"
+
+	// LBaaS
+	resourcePrefixLBaaSLoadBalancer = "/api/LBaaS/v1/loadbalancer.json"
+	resourcePrefixLBaaSFrontend     = "/api/LBaaS/v1/frontend.json"
+	resourcePrefixLBaaSBackend      = "/api/LBaaS/v1/backend.json"
+	resourcePrefixLBaaSServer       = "/api/LBaaS/v1/server.json"
+	resourcePrefixLBaaSBind         = "/api/LBaaS/v1/bind.json"
+	resourcePrefixLBaaSACL          = "/api/LBaaS/v1/ACL.json"
+
+	// VLAN
+	resourcePrefixVlan = "/api/vlan/v1/vlan.json"
+)
+
+type MockAPI interface {
+	api.API
+
+	OperationPathRequestCount(op types.Operation, path string) int
+}
+
+type mockAPI struct {
+	defaultAPI                   api.API
+	mockServer                   *ghttp.Server
+	resourceStorage              map[string]map[string][]byte
```

Comment:
I think `map[string]types.Object` should work (:tm:), this way you wouldn't need URLs at all